sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -711,12 +711,16 @@ class SparkSession private(
   // scalastyle:on
 
   /**
-   * Stop the underlying `SparkContext`.
+   * Stop the underlying `SparkContext` if there are are no active sessions remaining.
    *
    * @since 2.0.0
    */
   def stop(): Unit = {
-    sparkContext.stop()
+    if (SparkSession.numActiveSessions.get() == 0) {
```

Comment:
Shouldn't this remove itself from the active sessions before checking that there are 0 remaining?

Reply:
ah, good catch